### PR TITLE
feat(grouping): Add BuiltInFingerprintVariant [BE]

### PR DESCRIPTION
**Pull request by vartec**
Add BuiltInFingerprintVariant  to allow UI to differentiate between built-in and user-defined custom fingerprint rules
Frontend split into https://github.com/getsentry/sentry/pull/63402

implements: #63044

### CODE REVIEW

File: src/sentry/grouping/variants.py [hunk 1]
```diff
@@ -153,6 +153,16 @@ def _get_metadata_as_dict(self):
...
 
 
+class BuiltInFingerprintVariant(CustomFingerprintVariant):
+    """A built-in fingerprint."""
+
+    type = "built-in-fingerprint"
+
+    @property
+    def description(self):
+        return "built-in fingerprint"
```
Comment (scefali) [inline src/sentry/grouping/variants.py:163]: @vartec I think we wanted to say `Sentry Defined Fingerprint`

**Review by armenzg — CHANGES_REQUESTED** [review]
The PR looks good. We need to separate it into two different PRs before we can merge them.

File: src/sentry/grouping/fingerprinting/__init__.py [hunk 2]
```diff
@@ -237,6 +237,26 @@ def from_config_string(s, bases=None):
...
+        for r in fingerprinting_rules.rules:
+            r.is_builtin = True
+        return fingerprinting_rules
+
+    @classmethod
+    def _from_config_structure(cls, data, bases=None):
+        fingerprinting_rules = super()._from_config_structure(data, bases=bases)
+        for r in fingerprinting_rules.rules:
+            r.is_builtin = True
+        return fingerprinting_rules
```
[the remark this answers is not in the document]
Comment (vartec) [reply src/sentry/grouping/fingerprinting/__init__.py:257]: fully covered now.
Screenshot 2024-01-18 at 6 49 34 PM

**Review by armenzg — APPROVED** [review]
Great job @vartec !

**Review by jangjodi — APPROVED** [review]

**Pull request merged**